### ISSUE TITLE
`nameko` CLI called without any args fails

### ISSUE THREAD

**Issue by vlcinsky** [nameko/nameko#618]
**Describe the bug**
When `nameko` command is run without any argument, it fails with traceback.
**To Reproduce**
```bash
$ nameko
Traceback (most recent call last):
  File "/home/javl/sandbox/playwith_nameko/test_roman/.venv/bin/nameko", line 11, in <module>
    sys.exit(main())
  File "/home/javl/sandbox/playwith_nameko/test_roman/.venv/lib/python3.6/site-packages/nameko/cli/main.py", line 153, in main
    setup_config(args)
  File "/home/javl/sandbox/playwith_nameko/test_roman/.venv/lib/python3.6/site-packages/nameko/cli/main.py", line 134, in setup_config
    if args.config:
AttributeError: 'Namespace' object has no attribute 'config'
```

**Expected behavior**
`nameko` called without any arg shall show help or list names of all supported subcommands.

**Environment (please complete the following information):**
 - Nameko version: 3.0.0.rc6
 - Python version: 3.6
 - OS: Debian 9.7

**Additional context**
If you would accept using `click`, I would volunteer to rewrite existing `argparse` based CLI to `click` what would resolve reported issue.

**Comment by mattbennett** [comment]
@vlcinsky thank you for raising this.

We should definitely be returning a graceful message here, rather than this obtuse error message. I have no fond feelings for `argparse` and wouldn't mind adding a dependency on `click`.

**Comment by vlcinsky** [comment]
I will try to do quick rewrite to click and we will see.

**Comment by iky** [comment]
@vlcinsky great! Are you going to base your work on v3 branch?

**Comment by vlcinsky** [comment]
@iky yes, I branched from `v3.0.0-rc6` ignoring `global-config` one. If I shall rebase to `global-config`, i can do that.

**Comment by iky** [comment]
@vlcinsky `v3.0.0` branch is perfect, it has most of global config in already so you are fine with that one .) 

(`v3.0.0-rc6` tag is head of `v3.0.0` right now, all good here too)

**Comment by vlcinsky** [comment]
I have locally drafted CLI (without actually doing any real work) and have few notes:

`--config` and `--define` (plus possibly `--broker`) could be moved to the root command `nameko` as they are shared by all subcommands (for calls as `$ nameko --define AMQP_URI=pyampq://localhost --config config.yaml run --backdoor-port 9999`), but this would break backward compatibility so i will keep it where it is currently.

`nameko backdoor` allows to set `--rlwrap` or `--no-rlwrap`. In code I see checking for value None, but I assume expected statuses are True and False, defaulting to False and None shall be handled as False. If None has special meaning, there is no problem to return None (if no option is set) and True/Falso for option being present in the call.

My code is not committed yet, I shall do that within two or three days.

**Comment by vlcinsky** [comment]
WIP code pushed to https://github.com/vlcinsky/nameko/commit/6e0f3b1e1ead086a042166d5a8cadbcfd58fb30f and the command is temporarily called `cnameko`.

Assumes (just for printing values) python 3.6+, this restriction will be soon removed.

**Comment by vlcinsky** [comment]
I am in process of rewriting and have learned a lot from existing code. The test suite is impressive.

One of problems I have is with https://github.com/nameko/nameko/blob/v3.0.0-rc6/test/cli/test_main.py#L42

```python
from nameko.cli.main import main
import pytest
from mock import patch

@pytest.yield_fixture(autouse=True)
def fake_argv(empty_config):
    with patch.object(
        sys,
        'argv',
        [
            'nameko',
            'run',
            '--define',
            'AMQP_URI=pyamqp://someuser:*****@somehost/',
            'test.sample:Service',
        ],
    ):
        yield


def test_run():
    with patch('nameko.cli.run.main') as run:
        main()  # my code stays here retrying to connect
    assert run.call_count == 1
    (args,), _ = run.call_args
    assert args.define == [('AMQP_URI', 'pyamqp://someuser:*****@somehost/')]
```
The `test_run` runs the command via `main` (using faked command line parameters) and checks, if `nameko.cli.run:main` was called.

The problem I have is that my version calls the main (main CLI controller), but it really runs and does not complete as it keeps retrying connecting to RabbitMQ which is not accessible at the moment.

Existing test with existing code completes and succeeds and I have no idea how is that possible.

How it comes, that in existing nameko code the test runs through without running forever retrying connection to RabbitMQ?

**Comment by vlcinsky** [comment]
Resolved by (accepted) PR #620
